src/components/layout/layout.js

```diff
@@ -7,19 +7,98 @@ import Search from '../search/search';
 import TVShow from '../tvShow/tvShow';
 import Footer from './footer';
 import NotFound from '../notFound';
+import FavouriteContainer from '../favourites/favouriteContainer';
+import favouriteRepository from '../../repository/favouriteRepository';
 
-const Layout = () => (
-  <div>
-    <Menu />
-    <Switch>
-      <Route exact path="/" component={Home} />
-      <Route path="/search/:query" component={Search} />
-      <Route path="/search" component={Search} />
-      <Route path="/show/:showId" component={TVShow} />
-      <Route component={NotFound} />
-    </Switch>
-    <Footer />
-  </div>
-);
+class Layout extends Component {
+  constructor() {
+    super();
+    this.state = {
+      favourites: [],
```

Comment:
Nie podoba mi się że wyciągnęliśmy obsługę favorites aż tak wysoko do `Layout` - czyli komponentu który sobie wyobrażałem, że pozostanie w miarę prosty i będzie głównie zbiorem `routes` oraz czymś co narzuca podstawową strukturę html.
To jest do przegadania na żywo jak można to zaimplementować inaczej.
Potem jak będziemy dopinać reduxa to warto unikać podpinania pod store tego najwyższego komponentu, (blisko roota aplikacji), ponieważ wszelkie zmiany w `store` mogą triggerować rerendery tego komponentu, co pociągnie za sobą rerendery dzieci, a dzieci layoutu to praktycznie cała aplikacja ;)